### ISSUE TITLE
[ProjectModel.Server] DependencyDiagnostics even should come after Dependencies event

### ISSUE THREAD

**Issue by abpiskunov** [dotnet/cli#2420]
## Steps to reproduce

just check any Project.Model.Server trace, it sends info: 
```
Microsoft.DotNet.ProjectModel.Server.ProcessingQueue[0]
      OnSend ((750e884b62984f98b12a408a1b390000, DependencyDiagnostics, 2) -> {
        "Framework": {
          "FrameworkName": ".NETCoreApp,Version=v1.0",
          "FriendlyName": ".NETCoreApp,Version=v1.0",
          "ShortName": "netcoreapp1.0",
          "RedistListPath": null
        },
        "Errors": [],
        "Warnings": []
      })
```
before new Dependencies snapshot and this some times breaks tooling behavior for suppressing errors when restore is pending or running. For example, tooling determines if auto-restore is needed based on comparison of old and new dependencies snapshots. So when new unresolved dependencies are added we should show errors only after restore happens and is unable to restore unresolved dependencies. But if diagnostics come before Dependencies snapshot we can not guess where we before restore or after restore happened. 

This was caught by some random restore test failures , which caught unexpected errors showing during restore.

## Expected  behavior

Dependencies sent first, then DependenciesDiagnostics.

## Environment data
`dotnet --info` output:

C:\Windows\system32>dotnet --info
.NET Command Line Tools (1.0.0-beta-002275)

Product Information:
 Version:     1.0.0-beta-002275
 Commit Sha:  4af7437aa6

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10586
 OS Platform: Windows
 RID:         win10-x64

C:\Windows\system32>

**Comment by piotrpMSFT** [comment]
@pakrym, @anurse please take a look.

**Comment by troydai** [comment]
I assign this to myself. It's a quick fix.

**Comment by piotrpMSFT** [comment]
any update?

**Comment by troydai** [comment]
PR is out (#2441), waiting on CI. I'm drafting the mail to ask for merge permission from .net core ship room.

**Comment by troydai** [comment]
Fixed in 065e8352fc88aa2c1653d2321d39b1ab4efe5316
